### ISSUE TITLE
[TN-270] Account for manual changes to permissions/roles

### ISSUE THREAD

**Issue by ccrawford4** [human-internet/discord-bot#118]

**Comment by kofimokome** [comment]
First, we need to figure out if Discord provides events for:
1.  When the role of use changes.
2. When a user leaves the Discord server
3. When a user is removed from the Discord server.
All we need to do is to subscribe to these events and update the database when they are triggered.

**Comment by bastian-purrer** [comment]
@ccrawford4 yeah definitely 2 tickets, one for "role of user changes", one for "role is removed".
I don't see why your points c and d are needed though. What's the user impact scenario of not doing that?

**Comment by ccrawford4** [comment]
> @ccrawford4 yeah definitely 2 tickets, one for "role of user changes", one for "role is removed". I don't see why your points c and d are needed though. What's the user impact scenario of not doing that?
> 
> cc @kofimokome @Lunasocute @kyley422

I am not following - "user impact scenario of not doing that?" What do you mean? If a user leaves the server they should be considered un-verified right? I would imagine its the equivalent of if someone logs out of something - they should be required to log back in. By "leave the server" I mean a user actually removes their account from the server, not just exists discord. Same thing should apply if a server admin manually removes a user.

**Comment by bastian-purrer** [comment]
> If a user leaves the server they should be considered un-verified.
I disagree. We're not the login (that's discord), we're the humanity-check. why do that again?

**Comment by kyley422** [comment]
> At present if a server admin changes the role of a member this is not reflected in the Discord-Bot's db. For example: If a server admin removes the humanID verified role from a user then this should un-verify the user (requiring them to verify again) but it doesn't.
> 
> Steps to improve:
> 
> 1. Account for manual verification:
>    a. If a server admin adds the humanID Verified role to a user - this should be reflected in the Discord-Bots db
>    b. If a server admin removes the humanID Verified role for a user - they should be considered un-verified on our end as well
> 
> Sub-tasks (thank you kofi):
> 
> 1. Find out if Discord provides events for:
>    a. When the role of a user changes.
> 2. (If possible): Subscribe to these events and update the database when they are triggered

Re: Kofi's subtasks:
All of these are possible, just need to work on connecting the listener in discordbot.py to the database.

**Comment by bastian-purrer** [comment]
Status 7/29:
@bdonyan & @suganeswar03 - please both submit your solutions by Wednesday. 
Timeline:
Together under @Lunasocute 's leadership to combine so we can merge by EOW

**Comment by bdonyan** [comment]
submitted my solutions a few hours ago with video demonstration, ready for approval

**Comment by suganeswar03** [comment]
Even I have completed the ticket bastian, you can check in our tech group
in discord.

**Comment by bastian-purrer** [comment]
@Lunasocute Live by end of the week
